### PR TITLE
[MM-52821] Migrate components/app.jsx to Typescript

### PR DESCRIPTION
**Pull request by ilies-bel**
#### Summary

It seems that the file only needed renaming
- it does not have any associated tests
- does not use typed variables

#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/23439
https://mattermost.atlassian.net/browse/MM-52821

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/webpack.config.js [hunk 1]
```diff
@@ -384,7 +384,7 @@ async function initializeModuleFederation() {
 
     // Desktop specific code for remote module loading
     moduleFederationPluginOptions.exposes = {
-        './app': 'components/app.jsx',
+        './app': 'components/app.tsx',
```
Comment (M-ZubairAhmed) [inline webapp/channels/webpack.config.js:387]: we dont need to specify `tsx` extension here

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: webapp/channels/webpack.config.js (hunk 1 above)
Comment (ilies-bel) [reply webapp/channels/webpack.config.js]: That's right, thanks

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by hmhealey — APPROVED** [review]

**Comment by hmhealey** [comment]
Huh, I would've thought there was more to change in that file, but I guess we've moved stuff out of that file over time. Regardless, it's good to have it moved to TS though, so thanks for doing that!

**Comment by hmhealey** [comment]
Skipping QA review on this since there's no functional changes here

**Comment by hmhealey** [comment]
The EE tests are failing because we renamed this repo. I'll try updating to see if that fixes them

/update-branch

**Comment by hmhealey** [comment]
@ilies-bel Just a heads up, this PR is stuck due to a bug in our CI pipeline caused by us renaming this repository from `mattermost-server` to `mattermost` a couple weeks ago. One part of it is trying to clone https://github.com/ilies-bel/mattermost-server, but that doesn't exist because your fork is at https://github.com/ilies-bel/mattermost

We can just wait until the CI pipeline is fixed, but if you want, we might be work around that issue if you were to rename your fork to `mattermost-server` and then back to `mattermost` again. Doing that will make both URLs point to your repository since GitHub is good with redirecting when things are renamed, so it should let our CI pipeline do its work

**Comment by ilies-bel** [comment]
> @ilies-bel Just a heads up, this PR is stuck due to a bug in our CI pipeline caused by us renaming this repository from `mattermost-server` to `mattermost` a couple weeks ago. One part of it is trying to clone https://github.com/ilies-bel/mattermost-server, but that doesn't exist because your fork is at https://github.com/ilies-bel/mattermost
> 
> We can just wait until the CI pipeline is fixed, but if you want, we might be work around that issue if you were to rename your fork to `mattermost-server` and then back to `mattermost` again. Doing that will make both URLs point to your repository since GitHub is good with redirecting when things are renamed, so it should let our CI pipeline do its work

I just renamed, tell me if you need anything else

**Comment by hmhealey** [comment]
Thanks! That did successfully trick it into running, so that should be it once the E2E smoke testing completes

**Pull request merged**